source/Graphite.System/EventlogListener.cs

```diff
@@ -119,8 +125,19 @@ private void OnEntryWritten(object sender, EntryWrittenEventArgs e)
             if (!string.IsNullOrEmpty(this.category) && !this.category.Equals(e.Entry.Category, StringComparison.OrdinalIgnoreCase))
                 return;
 
+            if (this.eventid.HasValue && !this.eventid.Equals(e.Entry.EventID))
+            {
+                return;
+            }
+
+            var key = this.key;
+            if (this.attacheventidtokey)
+            {
+                key = string.Format("{0}.{1}", this.key, e.Entry.EventID);
+            }
+
             this.channel.Report(
-                this.key.Replace("${type}", typeTranslation[e.Entry.EntryType]),
+                key.Replace("${type}", typeTranslation[e.Entry.EntryType]),
```

Comment:
I'd suggest adding a call to `.Replace("${eventId}"...` here, instead of introducing the additional configuration option `AttachEventIDToKey`. That would be more flexible as well.